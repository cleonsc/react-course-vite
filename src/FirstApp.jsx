import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ title, subTitle, firsName }) => {

  return (
    <>
      <h1> {title} </h1>
      <p>{subTitle}</p>
      <p>{firsName}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
};

// Estos cargan antes que los proptypes por eso no da errors
FirstApp.defaultProps = {
  firsName: 'CÃ©sar',
  subTitle: 'No hay subtitulo',
  // title: 'No hay titulo',
}

export default FirstApp;
