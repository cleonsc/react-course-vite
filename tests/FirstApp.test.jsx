import React from 'react';
import { render } from '@testing-library/react';
import FirstApp from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
  test('should match with the snapshot', () => {
    const title = 'Hola, soy Goku';

    // Renderizamos el componente:
    const { container } = render(<FirstApp title={title} />);

    // compara contra una foto que se guarda en __snapshots__
    expect(container).toMatchSnapshot();
  })

  test('should show a title in  and h1 tag ', () => {
    const title = 'Hola, soy Goku';

    // Renderizamos el componente:
    const { container, getByText } = render(<FirstApp title={title} />);
    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector('h1')
    expect(h1.innerHTML).toContain(title)
  })
})
