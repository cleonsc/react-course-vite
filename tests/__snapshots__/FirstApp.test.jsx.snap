// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    CÃ©sar
  </p>
</div>
`;
